extensions-core/azure-extensions/src/main/java/org/apache/druid/storage/azure/AzureStorage.java

```diff
@@ -69,42 +73,41 @@ public class AzureStorage
    *
    * See OmniDataSegmentKiller for how DataSegmentKillers are initialized.
    */
-  private final Supplier<CloudBlobClient> cloudBlobClient;
+  private final Supplier<BlobServiceClient> blobServiceClient;
+  private final AzureClientFactory azureClientFactory;
+  private final ConcurrentHashMap<String, Pair<Integer, BlobContainerClient>> blobContainerClients = new ConcurrentHashMap<>();
 
   public AzureStorage(
-      Supplier<CloudBlobClient> cloudBlobClient
+      Supplier<BlobServiceClient> blobServiceClient,
```

Comment:
i think i can definitely just remove this supplier and lazily call factory.create instead, but i would prefer to make that as a followup change.